packages/alert-dialog/src/AlertDialog.ts

```diff
@@ -0,0 +1,92 @@
+/*
+Copyright 2023 Adobe. All rights reserved.
+This file is licensed to you under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License. You may obtain a copy
+of the License at http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software distributed under
+the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+OF ANY KIND, either express or implied. See the License for the specific language
+governing permissions and limitations under the License.
+*/
+import {
+    CSSResultArray,
+    html,
+    TemplateResult,
+} from '@spectrum-web-components/base';
+import { property } from '@spectrum-web-components/base/src/decorators.js';
+import { Dialog } from '@spectrum-web-components/dialog/src/Dialog.js';
+import '@spectrum-web-components/button/sp-button.js';
+import alertStyles from './alert-dialog.css.js';
+
+export type AlertDialogVariants =
+    | 'confirmation'
+    | 'information'
+    | 'warning'
+    | 'error'
+    | 'destructive'
+    | 'secondary'
+    | 'scroll'
+    | '';
+
+export const alertDialogVariants: AlertDialogVariants[] = [
+    'confirmation',
+    'information',
+    'warning',
+    'error',
+    'destructive',
+    'secondary',
+    'scroll',
+];
+
+export class AlertDialog extends Dialog {
```

Comment:
One side effect of extending from Dialog is its attribute/properties are included in the API table: https://feature-new-alertdialog--spectrum-web-components.netlify.app/components/alert-dialog/api/

Reply:
True if I create the Alert Dialog from scratch there will be a lot of repetition of code from Dialog. Either we make it DRY or we can abstract it and make it a standalone component detached from Dialog.

Reply:
I'm less afraid of code duplication than I am of bad docs. Hopefully the `@private` note allows us to get the best of both. 🤞🏼

Reply:
@private or @ignore doesnt work when are extending classes. I think we might need to fallback to use SpectrumElement and define our properties for Alert Dialog

Reply:
Another option here is to invert the class hierarchy. Alert Dialog is the less featured element, copy the things from it that are share to it and then extend it in Dialog?

Reply:
So this way Dialog will extend these Alert Dialog variants. Is this acceptable? This way we are not creating sp-alert-dialog.

Reply:
Just inverting the class hierarchy, not changing what elements are being defined. Roughly:
```
class AlertDialog extends SpectrumElement {
  // Add functionality shared between AlertDialog and Dialog here...
}

customElements.define('sp-alert-dialog', AlertDialog);

// ... other package

class Dialog extends AlertDialog {
  // Additional and cusotmized Dialog functionality here...
}

customElements.define('sp-dialog', Dialog);
```

Is that more clear?

Reply:
Updated. Let me know how this looks now.!